### PR TITLE
Integrate pitch

### PR DESCRIPTION
**Pull request by scottmacphersonmusic**
Aiming to let MusicTheory::Note accept injection of a Pitch object to derive frequency.

### CODE REVIEW

**Comment by bion** [comment]
careful with those @s, should probably wrap in backticks (these things: `) to avoid sending notifications to other github users

**Comment by bion** [comment]
i'm not clear on where you're getting nil. instead of commenting here, try clicking the 'Files changed' tab near the top of the page and comment directly on the line you're talking about.

File: test/implement_pitch_test.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'test_helper'
+
+class TestNote < MiniTest::Test
+  def setup
+    @pitch_object_note = MusicTheory::Note.new(frequency: Pitch.new(:B, 4), duration: 1.0, distort: false)
+    @frequency_note    = MusicTheory::Note.new(frequency: 493.883, duration: 1.0, distort: false)
```
Comment (bion) [inline test/implement_pitch_test.rb]: a pitch is not a frequency, which confuses this interface a bit. perhaps `pitch: Pitch.new(:B, 4)` instead

Comment (bion) [inline test/implement_pitch_test.rb]: and then internally you can decide what to do based on which key is present

File: test/implement_pitch_test.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  end
+
```
Comment (scottmacphersonmusic) [inline test/implement_pitch_test.rb]: THIS IS AN INLINE COMMENT!

File: lib/music_theory/note.rb [hunk 3]
```diff
@@ -53,6 +53,19 @@ def distort!(samples)
...
-  end
+    private
 
+    def interpret_frequency_or_pitch_object(frequency_option)
+      if frequency_option.class == Pitch
+        puts "its a Pitch object!!"
+        return frequency_option.frequency
+      elsif frequency_option == Float
+        return frequency_option
+      elsif frequency_option == Integer
```
Comment (bion) [inline lib/music_theory/note.rb]: this and line `62` are missing the call to `#class`, might be related to your bug with `nil`

File: test/implement_pitch_test.rb (hunk 2 above)
Comment (bion) [inline test/implement_pitch_test.rb]: yep, sure is

File: test/implement_pitch_test.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require 'test_helper'
+
+class TestPitchImplementation < MiniTest::Test
+  def setup
+    @pitch_note_B4      = MusicTheory::Note.new(pitch: Pitch.new(:B, 4), duration: 1.0, distort: false)
```
Comment (scottmacphersonmusic) [inline test/implement_pitch_test.rb:5]: @bion Do you think injecting a Pitch object is a good way to do this or would it be better to take an array as an argument 

`MusicTheory::Note.new(pitch: [:B, 4], duration: 1.0, distort: false)`

and write a wrapper to turn that into a Pitch object and extract its frequency inside MusicTheory::Note?

Comment (bion) [inline test/implement_pitch_test.rb:5]: In order of how loosely coupled they'll be:
1) `Note` doesn't know anything about `Pitch`:

``` ruby
MusicTheory::Note.new(frequency: Pitch.new(:B, 4).frequency)
```

2) `Note` knows how to ask `Pitch` for `frequency`: 

``` ruby
MusicTheory::Note.new(pitch: Pitch.new(:B, 4))
```

3) `Note` knows how to construct `Pitch` from a specially formatted raw data structure and ask it for `frequency`:

``` ruby
MusicTheory::Note.new(pitch: [:B, 4])
```

Comment (bion) [inline test/implement_pitch_test.rb]: - [ ] do the thing

**Pull request merged**